evm/src/main/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessor.java

```diff
@@ -136,7 +139,10 @@ private void exceptionalHalt(final MessageFrame frame) {
    *
    * @param frame The message frame
    */
-  protected void revert(final MessageFrame frame) {
+  protected void revert(final MessageFrame frame, final OperationTracer operationTracer) {
+    if (operationTracer != null) {
+      operationTracer.traceContextExit(frame);
+    }
```

Comment:
Note that in line 147 the state is set to `COMPLETED_FAILED` so the context exit will be called twice.